script/openqa-clone-custom-git-refspec

```diff
@@ -130,14 +130,13 @@ Please try 'curl $json_url' or select another job, e.g. in the same scenario: $h
         needles_dir="${needles_dir:-"$old_productdir/needles"}"
     fi
     local repo_branch="${repo_branch:-"$repo_name#$branch"}"
-    local test="${test:-"$testsuite@$repo_branch"}"
```

Comment:
i am thinking that only the expansion needs to be removed. `TEST` can be used to give a custom name (which happens to have use in some cases). Then i assume you have to check somehow if is set and only then add it to the `$cmd`. WDYT?

Reply:
So `TEST` should not be altered by default but setting the environment variable `test` should still enable one to set it?

> Then i assume you have to check somehow if is set and only then add it to the $cmd. WDYT?

That needed to be done, otherwise we would delete the `TEST` setting by default.

I'd say it makes more sense to simply allow specifying additional, arbitrary parameters to append to the clone invocation (instead of making a special case for `TEST`).

Reply:
And note that this already works, e.g. 
```
script/openqa-clone-custom-git-refspec --dry-run https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/14495 https://openqa.suse.de/tests/8291973 BUILD=b10n1k/os-autoinst-distri-opensuse#14495 TEST=foo SOME=setting```
invokes
```
script/openqa-clone-job --skip-chained-deps --parental-inheritance --within-instance https://openqa.suse.de 8291973 _GROUP=0 BUILD=b10n1k/os-autoinst-distri-opensuse#14495 CASEDIR=https://github.com/b10n1k/os-autoinst-distri-opensuse.git#hpc_scipy PRODUCTDIR=os-autoinst-distri-opensuse/products/sle NEEDLES_DIR=sle/products/sle/needles BUILD=b10n1k/os-autoinst-distri-opensuse#14495 TEST=foo  SOME=setting
```

I think that should be good enough to set arbitrary settings. Of course you still need to keep in mind that *all* jobs will have that setting so it still doesn't make much sense to set e.g. `TEST` for jobs with dependencies.

Reply:
Is there an alternative right now to override the name? I'm honstly not sure it's documented anywhere and I can see your argument with regard to inconsistencies. I would consider it common practice, though, and we should at least think of other options before breaking that use case.

Reply:
FTR we discussed it briefly in the daily. The most practical option might be to support +TEST here (maybe not in this PR)